modules/integration/src/test/scala/scala/cli/integration/SipScalaTests.scala

```diff
@@ -515,6 +515,39 @@ class SipScalaTests extends ScalaCliSuite with SbtTestHelper with MillTestHelper
     }
   }
 
+  test("consecutive -Wconf:* flags are not ignored") {
+    val sv                 = "3.5.2"
+    val sourceFileName     = "example.scala"
+    val warningConfOptions = Seq("-Wconf:cat=deprecation:e", "-Wconf:any:s")
+    TestInputs(os.rel / sourceFileName ->
+      s"""//> using scala $sv
+         |//> using options ${warningConfOptions.mkString(" ")}
```

Comment:
This test checks the behaviour for repeated options added as a directive. Do/can we have a test to verify this works the same for options passed via command line (I checked manually that the behaviour isn't consistent currently, without the changes from this PR)

Reply:
...I get the feeling that's overkill, but I guess it won't hurt. 
Let me add that.

Reply:
turns out it wasn't just overkill, but uncovered another bug.
Added a test + options from both command line and directives should now follow the same ordering (they weren't)